src/components/buttons/liquidity_confirmation_button/index.tsx

```diff
@@ -1,38 +1,23 @@
 import Button, { ButtonProps as MuiButtonProps } from '@material-ui/core/Button'
-import DoneIcon from 'assets/icons/DoneIcon.svg'
 import React, { Dispatch, SetStateAction } from 'react'
 import { colors } from 'utils/mui'
 import { useStyles } from './styles'
 
 interface ButtonProps extends MuiButtonProps {
   text: string
-  activateWeb3Account?: () => void
-  activeWeb3Account?: string
-  confirmed?: boolean
-  setConfirmed?: Dispatch<SetStateAction<boolean>>
+  approved?: boolean
+  darkType?: boolean
+  setApproved?: Dispatch<SetStateAction<boolean>>
 }
 
 const ConfirmationButton = (props: ButtonProps) => {
   const { buttonStyles } = useStyles()
-  const { text, activateWeb3Account, activeWeb3Account, disabled, onClick, confirmed, setConfirmed } = props
+  const { text, onClick, approved, setApproved, darkType, disabled } = props
 
-  const buttonBgColor = disabled
-    ? colors.primaryButtonBgDisabled
-    : confirmed
-    ? 'transparent'
-    : colors.primaryButtonBgColor
-  const buttonTextColor = disabled
-    ? colors.walletButtonBorderColor
-    : confirmed
-    ? colors.primaryButtonBgColor
-    : colors.primaryButtonTextColor
+  const buttonBgColor = darkType ? colors.voteButtonDarkBg : disabled ? 'transparent' : colors.primaryButtonBgColor
+  const buttonTextColor = darkType || disabled ? colors.white : colors.primaryButtonTextColor
 
-  const handleClick =
-    !activeWeb3Account || activeWeb3Account.length === 0
-      ? activateWeb3Account
-      : activeWeb3Account && activeWeb3Account.length > 0 && setConfirmed
-      ? () => setConfirmed(!confirmed)
-      : onClick
+  const handleClick = setApproved ? () => setApproved(!approved) : onClick
```

Comment:
Is it easier to pass only onClick there? For example, if you wanna use this button not only for confirmation